ec/ecresource/deploymentresource/expanders.go

```diff
@@ -133,10 +135,15 @@ func updateResourceToModel(d *schema.ResourceData, client *api.API) (*models.Dep
 		unsetTopology(es)
 	}
 
+	// TODO: There's potential for problems when a user upgrades a deployment
+	// with Elasticsearch curation settings (Pre-ILM) to a version >= 6.6.0,
+	// the Elasticsearch curation settings are removed without migrating these
+	// calling the "enable-deployment-resource-ilm" API. This needs improvement.
```

Comment:
Potentially, we don't need to improve this if we consider that we're already enabling version 6.6.x or lower to work.